### ISSUE TITLE
Type Error in Base Code

### ISSUE THREAD

**Issue by DB6CP** [dfigus/addon-solarflow-control#40]
Possible Problem in the Base Code from Reinhard:

https://github.com/reinhard-brandstaedter/solarflow-control/issues/204#issuecomment-2553155565

TypeError: unsupported operand type(s) for -: 'int' and 'NoneType'
[09:51:49] INFO: Running Solarflow Control
Traceback (most recent call last):
File "/solarflow/solarflow-control.py", line 82, in
or int(os.environ.get('MAX_INVERTER_INPUT',MAX_INVERTER_LIMIT - MIN_CHARGE_POWER))
~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~

Seems like MIN_CHARGE_POWER is None. 

Normaly that should be fetched from Config or not ?!

**Comment by dfigus** [comment]
Hi @DB6CP,

thanks for reporting. Please don't report issues in the hassio-addons repo, but instead in the individual repositories. That makes it easier (and I don't have notifications setup for hassio-addons 🙃).

Can you share your log from the addon startup, which version of the addon you're running and the configuration values?

**Comment by dfigus** [comment]
it looks like reinhard-brandstaedter/solarflow-control#314 introduced the new configuration `MAX_INVERTER_LIMIT` and this is probably `None`. Can you please try the latest 0.10.2? This adds the new configuration option.

**Comment by DB6CP** [comment]
currently fiddling with a docker setup for SF-Control. But yup will try 0.10.2

**Comment by dfigus** [comment]
I directly released 0.10.2, maybe you need to refresh in the HA addon view (top right button).

**Comment by DB6CP** [comment]
did just that. its already setup. Just waiting for my MQTT Broker to get back online (update) . Will report shortly

**Comment by DB6CP** [comment]
Seems to run after i set Lat and Lon. When omitting those i get : 
`
    LAT = config.getfloat('global', 'latitude', fallback=None) or float(os.environ.get('LATITUDE',0))
          ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 852, in getfloat
    return self._get_conv(section, option, float, raw=raw, vars=vars,
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                          fallback=fallback, **kwargs)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 837, in _get_conv
    return self._get(section, conv, option, raw=raw, vars=vars,
           ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                     **kwargs)
                     ^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 832, in _get
    return conv(self.get(section, option, **kwargs))
ValueError: could not convert string to float: ''
[16:38:01] INFO: Running Solarflow Control
Traceback (most recent call last):
  File "/solarflow/solarflow-control.py", line 105, in <module>
    LAT = config.getfloat('global', 'latitude', fallback=None) or float(os.environ.get('LATITUDE',0))
          ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 852, in getfloat
    return self._get_conv(section, option, float, raw=raw, vars=vars,
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                          fallback=fallback, **kwargs)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 837, in _get_conv
    return self._get(section, conv, option, raw=raw, vars=vars,
           ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                     **kwargs)
                     ^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 832, in _get
    return conv(self.get(section, option, **kwargs))
ValueError: could not convert string to float: ''
[16:38:05] INFO: Running Solarflow Control
Traceback (most recent call last):
  File "/solarflow/solarflow-control.py", line 105, in <module>
    LAT = config.getfloat('global', 'latitude', fallback=None) or float(os.environ.get('LATITUDE',0))
          ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 852, in getfloat
    return self._get_conv(section, option, float, raw=raw, vars=vars,
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                          fallback=fallback, **kwargs)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 837, in _get_conv
    return self._get(section, conv, option, raw=raw, vars=vars,
           ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                     **kwargs)
                     ^^^^^^^^^
  File "/usr/local/lib/python3.13/configparser.py", line 832, in _get
    return conv(self.get(section, option, **kwargs))
ValueError: could not convert string to float: ''
'

**Comment by dfigus** [comment]
The latitude and longitude where previously in the wrong section of the config file and hence all values were ignored. An empty value is not accepted by the base code (yielding the above error) and hence I updated the replace logic in case no coordinates are provided (fully optional for the addon). The newest version is being deployed to the edge repository now. Could you test the recent build there?

**Comment by DB6CP** [comment]
Started and looks good so far. Log is clean.

**Comment by dfigus** [comment]
I just released v0.10.3. Thanks again for reporting the issue! If you have any other wishes/ideas or just general feedback, just let me know!
